proto/structured.proto

```diff
@@ -27,61 +27,51 @@ option (gogoproto.sizer_all) = true;
 option (gogoproto.marshaler_all) = true;
 option (gogoproto.unmarshaler_all) = true;
 
-message Table {
-  optional string name = 1 [(gogoproto.nullable) = false];
+// For Cockroach internal use.
+message Internal {
+  optional uint32 id = 1 [(gogoproto.nullable) = false];
 }
-
+ 
 message Column {
   enum ColumnType {
     BYTES = 0;
   }
-
   optional string name = 1 [(gogoproto.nullable) = false];
   optional ColumnType type = 2 [(gogoproto.nullable) = false];
+  // For Cockroach internal use.
+  optional Internal internal = 9 [(gogoproto.nullable) = true];
 }
 
 message Index {
   optional string name = 1 [(gogoproto.nullable) = false];
   optional bool unique = 2 [(gogoproto.nullable) = false];
+  // An ordered list of column names of which the index is comprised. Each
+  // column_name refers to a column in the TableSchema's columns. Special
+  // care is taken to update this when deleting columns.
+  repeated string column_names = 3 [(gogoproto.nullable) = false];
+  // For Cockroach internal use.
+  optional Internal internal = 9 [(gogoproto.nullable) = true];
 }
 
+// TableSchema used to configure a db table. The internal messages
+// within Column, Index, and TableSchema are for cockroach internal use.
 message TableSchema {
-  message IndexByName {
-    optional Index index = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
-    // An ordered list of column names of which the index is comprised. Each
-    // column_name refers to a column in the TableSchema's columns.
-    repeated string column_names = 2 [(gogoproto.nullable) = false];
-  }
-  optional Table table = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional string name = 1 [(gogoproto.nullable) = false];
   repeated Column columns = 2 [(gogoproto.nullable) = false];
   // An ordered list of indexes included in the table. The first index is the
   // primary key; it is required.
-  repeated IndexByName indexes = 3 [(gogoproto.nullable) = false];
-}
-
-message ColumnDescriptor {
-  optional uint32 id = 1 [(gogoproto.nullable) = false];
-  optional Column column = 2 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
-}
-message IndexDescriptor {
-  optional uint32 id = 1 [(gogoproto.nullable) = false];
-  optional Index index = 2 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
-  // An ordered list of column ids of which the index is comprised. Each
-  // column_id refers to a column in the TableDescriptor's columns; special
-  // care is taken to update this when deleting columns.
-  repeated uint32 column_ids = 3 [(gogoproto.nullable) = false];
+  repeated Index indexes = 3 [(gogoproto.nullable) = false];
```

Comment:
Should we call these ColumnSchema and IndexSchema for consistency? Or remove the suffix Schema from TableSchema?